driver-core/src/main/com/mongodb/client/model/Filters.java

```diff
@@ -637,12 +672,14 @@ public SimpleEncodingFilter(final String fieldName, final TItem value) {
     private static <TItem> void encodeValue(final BsonDocumentWriter writer, final TItem value, final CodecRegistry codecRegistry) {
         if (value == null) {
             writer.writeNull();
+        } else if (value instanceof Bson) {
+            ((Encoder) codecRegistry.get(BsonDocument.class)).encode(writer,
+                    ((Bson) value).toBsonDocument(BsonDocument.class, codecRegistry), EncoderContext.builder().build());
         } else {
             ((Encoder) codecRegistry.get(value.getClass())).encode(writer, value, EncoderContext.builder().build());
         }
     }
 
-    // WIP, to be used later by not/nor methods
     private static class NotFilter implements Bson {
```

Comment:
It's not trying to do the same as the .NET version.  This code is solely a helper for creating `$not` queries and cannot be used to produce alternative query types.  The reasoning is that users explicitly chose which query type via the supplied helper.